SIL.Scripture/BookSet.cs

```diff
@@ -0,0 +1,321 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace SIL.Scripture
+{
+	/// <summary>
+	/// Traditionally Paratext stored selected sets of books as ascii strings
+	/// of 1's and 0's. This class wraps this representation to allow manipulating
+	/// and iterating through this structure.
+	/// </summary>
+	public sealed class BookSet
+	{
+		#region Fields and Xtors
+		private readonly bool[] selected = new bool[Canon.AllBookIds.Length];
+
+		// constant to simplify filtered operations code
+		public static readonly BookSet AllBooks = new BookSet(true);
+
+		// The empty BookSet.
+		public static readonly BookSet Empty = new BookSet();
+
+		public BookSet()
+		{
+		}
+
+		public BookSet(params string[] books)
+		{
+			if (books != null && books.Length == 1 && books[0].Length != 3)
```

Comment:
Unfortunately, there is no guarantee that the length of the string would exactly equal `selected.Length` (e.g. it could only contain the first 66 books).

Adding in another constructor that takes just a string doesn't work because then you can't do `new BookSet("MRK")` - unless you had that constructor also handle that case. It was just easier to just have one and have that one constructor have the smarts to determine what was meant.